onedal/datatypes/utils/common.cpp

```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
+* Copyright 2024 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <cstdint>
+
+namespace oneapi::dal::python {
+
+union endian_checker {
+    std::uint64_t blob;
+    std::uint8_t arr[8];
+};
```

Comment:
oneDAL works only on little-endian hardware.
Do we need this logic at all?

Reply:
Or this code is aimed to re-pack the input tensors from big-endian to little-endian?
In this case some comments would be helpful.

Reply:
Actually it was for checking hardware but was not designed well and used for saving metadata inf in `_sycl_usm_array_interface_["typestr"]`. Thank you for the information. I think it make sense just hardcode it with a comment.